modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/permission/test/DLFileEntryTypePermissionCheckerTest.java

```diff
@@ -0,0 +1,138 @@
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.security.auth.PrincipalThreadLocal;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.PermissionCheckerFactoryUtil;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.service.CompanyLocalServiceUtil;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.service.UserGroupRoleLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.security.permission.test.util.BasePermissionTestCase;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portlet.documentlibrary.constants.DLConstants;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Alicia García
+ */
+@RunWith(Arquillian.class)
+public class DLFileEntryTypePermissionCheckerTest
+	extends BasePermissionTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+
+		PermissionThreadLocal.setPermissionChecker(_originalPermissionChecker);
+
+		PrincipalThreadLocal.setName(_originalName);
+	}
+
+	@Test
+	public void testContains() throws Exception {
+		Company company = CompanyLocalServiceUtil.getCompany(
+			group.getCompanyId());
+
+		DLFileEntryType dlVideoExternalShortcutDLFileEntryType =
+			DLFileEntryTypeLocalServiceUtil.getFileEntryType(
+				company.getGroupId(), "DL_VIDEO_EXTERNAL_SHORTCUT");
+
+		Assert.assertTrue(
+			_dlFileEntryTypeModelResourcePermission.contains(
+				permissionChecker, dlVideoExternalShortcutDLFileEntryType,
+				ActionKeys.VIEW));
+
+		DLFileEntryType googleDocsDLFileEntryType =
+			DLFileEntryTypeLocalServiceUtil.getFileEntryType(
+				company.getGroupId(), "GOOGLE_DOCS");
+
+		Assert.assertTrue(
+			_dlFileEntryTypeModelResourcePermission.contains(
+				permissionChecker, googleDocsDLFileEntryType, ActionKeys.VIEW));
```

Comment:
Can you inject the services instead of using the *Util? You're injecting some, but then using others through the Util classes.

Reply:
good catch!!

Thanks for pointing at it